gravtr/__init__.py

```diff
@@ -1,14 +1,35 @@
 import hashlib
-
-GRAVATAR_URL = 'http://www.gravatar.com/avatar/'
+import sys
+if sys.version_info[0] < 3:
+    import urllib
+else:
+    import urllib.parse as urllib
 
 class Gravtr(object):
+    GRAVATAR_URL = 'https://www.gravatar.com/avatar/'
+    GRAVATAR_URL_UNSECURE = 'http://www.gravatar.com/avatar/'
+
+    class ratingType(object):
+        G = 'g'
+        PG = 'pg'
+        R = 'r'
+        X = 'x'
+
     def __init__(self, email):
-        self.email = email
+        self.email = email.encode('utf-8')
 
-    def generate(self, size=None):
-        email = self.email.encode('utf-8')
-        self.url = GRAVATAR_URL + hashlib.md5(email).hexdigest()
+    def generate(self, unsecure=False, size=None, typed=False, default=None, force_default=False, rating_type=None):
```

Comment:
Landscape / Codacy (well, really PyLint) is going to complain on about this line being too long.
Should we build I build it a different way or just add an exception?